tests/framework/artifacts.py

```diff
@@ -178,6 +322,40 @@ def _fetch_artifacts(self,
                                                 artifact_type=artifact_type))
         return artifacts
 
+    def snapshots(self, keyword=None):
+        """Return snapshot artifacts for the current arch."""
+        # Snapshot artifacts are special since they need to contain
+        # a variable number of files: mem, state, disks, ssh key.
+        # To simplify the way we retrieve and store snapshot artifacts
+        # we are going to group all snapshot file in a folder and the
+        # "keyword" parameter will filter this folder name.
+        #
+        # Naming convention for files within the snapshot below.
+        # Snapshot folder /ci-artifacts/snapshots/x86_64/fc_snapshot_v0.22:
+        # - vm.mem
+        # - vm.vmstate
+        # - disk0 <---- this is the root disk
+        # - disk1
+        # - diskN
+        # - ssh_key
+
+        artifacts = []
+        prefix = ArtifactCollection.ARTIFACTS_ROOT
+        prefix += ArtifactCollection.ARTIFACTS_SNAPSHOTS
+        snaphot_dirs = self.bucket.objects.filter(Prefix=prefix)
+        for snapshot_dir in snaphot_dirs:
+            key = snapshot_dir.key
+            # Filter out the snapshot artifacts root folder.
+            # Select only files with specified keyword.
+            if (key[-1] == "/" and key != prefix and
```

Comment:
Comparing `key[-1] with "/"` is a bit unclear. Is the purpose of this comparison to check if it's a folder?

Reply:
Yup, wanted to filter out the folder.